tooling/hibernate-gradle-plugin/src/main/groovy/org/hibernate/orm/tooling/gradle/HibernatePlugin.java

```diff
@@ -43,10 +44,29 @@ private void applyEnhancement(final Project project, final HibernateExtension hi
 
 			final Task compileTask = project.getTasks().findByName( sourceSet.getCompileJavaTaskName() );
 			assert compileTask != null;
-			compileTask.doLast(
-					task -> EnhancementHelper.enhance( sourceSet, hibernateExtension.enhance, project )
-			);
+			compileTask.doLast(new EnhancerAction( sourceSet, hibernateExtension, project ));
```

Comment:
I would suggest adding some comment explaining why lambda is not a viable solution (including the gradle bug link) so we can avoid confusion and raise awareness of when the lambda can be reverted back.

Reply:
Added a comment